### PR TITLE
DatePicker: Default Value should be undefined, not new Date()

### LINKED ISSUE #3574: DatePicker: Default value should be undefined, not new Date
### Bug Report

- __Package version(s)__: latest

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

#### Describe the issue:
DatePicker sets the default `selectedDate` to `new Date()` if no value is passed via props. Then, if the consumer passes the current date (`new Date()`) in props, the text box does not update because the `selectedDate` is the same as the value passed in props. This is unexpected, the default `selectedDate` should be undefined

### PR DESCRIPTION
**Pull request by abettadapur**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #3574 #3683
- [x] Include a change request file using `$ npm run change`

#### Description of changes

* `selectedDate` defaults to undefined in constructor and willReceiveProps
* `navigatedDate` state was removed because it was unused

### CODE REVIEW

**Comment by abettadapur** [comment]
@dzearing Could I get a review on this? Not sure who owns this component

**Comment by abettadapur** [comment]
@johannao76 @lorejoh12 Could I get a review on this one?

**Comment by nguyentoantuit** [comment]
@johannao76  @lorejoh12 If it is possible, please take a look at that. I got stuck because of this issue. My client really want this issue should be fixed. Thank a lot!

**Comment by cschleiden** [comment]
Is this a behavior change for users who rely on the default date being the current date today?

**Comment by abettadapur** [comment]
The datepicker still displays an empty input if no date is passed, so as I see it, the default date doesnt really serve a purpose

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -133,7 +131,7 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
     let { formatDate, value } = props;
 
     this.state = {
-      selectedDate: value || new Date(),
+      selectedDate: value || undefined,
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx]: Can't this change break some consumers that would expect the default to be new Date()? Seems like this should be controlled by a prop.

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -22,8 +22,6 @@ import * as stylesImport from './DatePicker.scss';
 const styles: any = stylesImport;
 
 export interface IDatePickerState {
-  /** The currently focused date in the drop down, but not necessarily selected */
-  navigatedDate?: Date;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx]: You are removing this from the interface, can this break some existing consumers of this code?

**Pull request merged**